scripts/lib/runner.ts

```diff
@@ -0,0 +1,112 @@
+import { strict as assert } from "assert";
+import { resolve, parse, sep } from "path";
+
+import chalk from "chalk";
+import { formatDistanceStrict } from "date-fns";
+
+import { initFirebaseAdminApp } from "./helpers";
+import {
+  log,
+  SCRIPT,
+  displayHelp,
+  log as actualLog,
+  displayProps,
+} from "./log";
+import { SimConfig, SimStats, LogFunction } from "./types";
+import { loopUntilKilled, readConfig } from "./utils";
+
+export const SIM_EXT = [".config.json5", ".config.json"];
+
+export type InitFirebaseOptions = {
+  log: LogFunction;
+  conf: SimConfig;
+  stats: SimStats;
+};
+
+export const initFirebase: (options: InitFirebaseOptions) => void = ({
+  log,
+  conf: { credentials, projectId },
+  stats,
+}) => {
+  log(
+    chalk`{inverse NOTE} Initializing Firebase with {green ${credentials}} for project {green ${projectId}}...`
+  );
+
+  assert.ok(
+    projectId,
+    chalk`${initFirebase.name}(): {magenta projectId} is required`
+  );
+
+  (stats.file ??= {}).credentials = credentials
+    ? resolve(process.cwd(), credentials)
+    : undefined;
+
+  initFirebaseAdminApp(projectId, {
+    credentialPath: stats.file.credentials,
+  });
+
+  log(
+    chalk`{green.inverse DONE} Initialized  Firebase for {green ${projectId}}.`
+  );
+};
+
+export const run: (
```

Comment:
`run` is utility function that sets up DB connection and similar objects for use by the `main` logic of the script and even calls a `cleanup` function afterwards, but is abstracted away with as little context of the actual script purpose so it can possibly be reused by other scripts.